src/data/int/parity.lean

```diff
@@ -33,9 +33,62 @@ theorem odd_iff {n : ℤ} : odd n ↔ n % 2 = 1 :=
 lemma not_even_iff {n : ℤ} : ¬ even n ↔ n % 2 = 1 :=
 by rw [even_iff, mod_two_ne_zero]
 
+lemma not_odd_iff {n : ℤ} : ¬ odd n ↔ n % 2 = 0 :=
+by rw [odd_iff, mod_two_ne_one]
+
+lemma even_iff_not_odd {n : ℤ} : even n ↔ ¬ odd n :=
+by rw [not_odd_iff, even_iff]
+
 @[simp] lemma odd_iff_not_even {n : ℤ} : odd n ↔ ¬ even n :=
 by rw [not_even_iff, odd_iff]
 
+lemma even_or_odd' (n : ℤ) : ∃ k : ℤ, n = 2 * k ∨ n = 2 * k + 1 :=
```

Comment:
Once you have `even_iff_not_odd`, `even_or_odd` is trivial. Then `even_or_odd'` immediately follows from the definitions of `even` and `odd`